### ISSUE TITLE
[Q] Can sslh transparent proxy mode work with stunnel?

### ISSUE THREAD

**Issue by roytam1** [yrutschle/sslh#209]
README doesn't show this use case, I wonder if this can be setup.

**Comment by yrutschle** [comment]
On Mon, Dec 17, 2018 at 10:05:49PM -0800, roytam1 wrote:
> README doesn't show this use case, I wonder if this can be setup.

I'm not sure what you're trying to achieve. stunnel does
TLS, so no problem for sslh to proxy an stunnel connection
transparently.

I'm not sure what you expect a transparent stunnel would do:
forward the local IP address to the final service?

**Comment by roytam1** [comment]
> forward the local IP address to the final service?

yes thats what I want to do

**Comment by yrutschle** [comment]
That's specifically inside the OpenVPN configuration then, and independant from sslh, which does not even need to run as transparent (the openvpn server will simply see the connection as coming from sslh, but it'll still see the origin IP)

Let's say you are connecting a 10.1.0.0 network to a 10.2.0.0 network over the Internet, that your public IPs are 192.0.2.1 and 198.51.100.1 respectively:

ASCII-art time:
```


10.1.0.23                                           10.2.1.3
     |                                                ^
     v    192.0.2.1               198.51.100.1        |
   openvpn1 <--------------------------> sslh ----> openvpn2
```

Here, openvpn2 will see its openvpn connection coming from sslh (if non-transparent), but that doesn't matter as far as final clients are concerned: 10.2.1.3 does communicate with 10.1.0.23 and should see that IP address.

Or, I'm confused as to what you're trying to do and you'll need to explain in more than a one-line sentence.

**Comment by roytam1** [comment]
> Or, I'm confused as to what you're trying to do and you'll need to explain in more than a one-line sentence.

My current situation is using stunnel server to serve SSL/TLS connections on port 443(so I can use proxytunnel or socat to encapsulate ssh connection into normal HTTPS to bypass corporate firewall and proxy in office) with sslh backend, and for example ssh shows the connected IP as 127.0.0.1 which I want to prevent.

**Comment by MattElekHarris** [comment]
I'm sorry for commenting on a relatively old issue, but I'm trying to get the same thing to work, to no avail. I'm trying to enable transparent proxying for both stunnel and SSLH, and they work independently, but not in tandem. I'm trying to use stunnel4 to decrypt HTTP and ssh traffic, and then SSLH to send them to their proper services. Both proxy transparently independently, but when I try to use them together, SSLH shows this error for every connection:
Jul 22 17:36:33 hostname sslh[3116]: bind:98:Address already in use
Jul 22 17:36:33 hostname sslh[3116]: bind_peer:98:Address already in use
Jul 22 17:36:33 hostname sslh[3116]: connect: Address already in use

I saw you said to use a different mark, and I tried that, but I have a very hard time understanding iptables so any suggestions or guidance would be greatly appreciated. Thanks in advance.
